impl/maven-core/src/main/java/org/apache/maven/internal/impl/DefaultTypeRegistry.java

```diff
@@ -55,11 +53,8 @@ public class DefaultTypeRegistry extends AbstractEventSpy implements TypeRegistr
     private final LegacyArtifactHandlerManager manager;
 
     @Inject
-    public DefaultTypeRegistry(
-            List<TypeProvider> providers, LanguageRegistry languageRegistry, LegacyArtifactHandlerManager manager) {
-        this.types = requireNonNull(providers, "providers cannot be null").stream()
-                .flatMap(p -> p.provides().stream())
-                .collect(Collectors.toMap(Type::id, identity()));
+    public DefaultTypeRegistry(Lookup lookup, LanguageRegistry languageRegistry, LegacyArtifactHandlerManager manager) {
+        this.lookup = lookup;
```

Comment:
nullable?

Reply:
AFAIK, nullable must be opt-in for injection, so no annotation here means the dependencies are required.

Reply:
I was more about requireNonNull ... As other ctor params